File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    
    stack = []
    for i in s:
        if i == "(":
            stack.append(i)
        else:
            if stack and stack.pop() == "(":
                continue
            answer = False
            break
    if stack:
        answer = False
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Character> stack = new Stack<>();
        
        for(int i = 0; i < s.length(); i++) {
            Character c = s.charAt(i);
            if(c == '(') stack.push(c);
            else if(c == ')') {
                if(!stack.isEmpty() && stack.peek() == '(') stack.pop();
                else {
                    answer = false;
                    break;
                }
            }
        }
        if(!stack.isEmpty()) answer = false;

        return answer;
    }
}